### PR TITLE
Add support for custom titlebar on Linux with Qt < 5.15

### PR DESCRIPTION
**Pull request by ilya-fedin**

### CODE REVIEW

File: Telegram/SourceFiles/platform/linux/specific_linux.cpp [hunk 1]
```diff
@@ -268,6 +278,142 @@ bool GenerateDesktopFile(
...
+	xev.type = moveResize;
+	xev.sequence = 0;
+	xev.window = window->winId();
+	xev.format = 32;
+	xev.data.data32[0] = globalPos.x();
+	xev.data.data32[1] = globalPos.y();
+	if (edges == 16)
+		xev.data.data32[2] = 8; // move
+	else
+		xev.data.data32[2] = XCBMoveResizeFromEdges(Qt::Edges(edges));
```
Comment (23rd) [inline Telegram/SourceFiles/platform/linux/specific_linux.cpp:371]:
```suggestion
	xev.data.data32[2] = (edges == 16)
		? 8 // move
		: XCBMoveResizeFromEdges(Qt::Edges(edges));
```

File: Telegram/SourceFiles/platform/linux/specific_linux.cpp [hunk 2]
```diff
@@ -268,6 +278,142 @@ bool GenerateDesktopFile(
...
+		false,
+		screen->root,
+		XCB_EVENT_MASK_SUBSTRUCTURE_REDIRECT | XCB_EVENT_MASK_SUBSTRUCTURE_NOTIFY,
+		reinterpret_cast<const char*>(&xev));
+
+	return true;
+}
+
+bool StartWaylandMove(QWindow *window) {
+	if (auto waylandWindow = static_cast<QtWaylandClient::QWaylandWindow*>(window->handle())) {
```
Comment (23rd) [inline Telegram/SourceFiles/platform/linux/specific_linux.cpp:386]: 92 characters? 😕

File: Telegram/SourceFiles/platform/linux/specific_linux.cpp (hunk 1 above)
Comment (ilya-fedin) [reply Telegram/SourceFiles/platform/linux/specific_linux.cpp]: fixed

File: Telegram/SourceFiles/platform/linux/specific_linux.cpp (hunk 2 above)
Comment (ilya-fedin) [reply Telegram/SourceFiles/platform/linux/specific_linux.cpp]: fixed

**Comment by john-preston** [comment]
@ilya-fedin looks fine, just conflicts need fixing.

**Comment by john-preston** [comment]
Thanks!

**Pull request merged**